File: Neon-Java/src/com/islandstudio/neon/Stable/secondary/iCommand/CommandSyntax.java
package com.islandstudio.neon.stable.secondary.iCommand;

import org.bukkit.ChatColor;

public enum CommandSyntax {
    INVALID_ARGUMENT(ICommand.COLORED_PLUGIN_NAME + ChatColor.RED + "Invalid or missing argument! Please type '/neon' to show available command syntax!"),
    INVALID_PERMISSION(ICommand.COLORED_PLUGIN_NAME + ChatColor.RED + "You don't have permission to use this command!");

    private final String syntaxMessage;

    CommandSyntax(String syntaxMessage) {
        this.syntaxMessage = syntaxMessage;
    }

    public String getSyntaxMessage() {
        return syntaxMessage;
    }

    public static class Handler {
        public static String createSyntaxMessage(String message) {
            return ICommand.COLORED_PLUGIN_NAME + message;
        }
    }
}

File: Neon-Kotlin/src/main/com/islandstudio/neon/stable/primary/nCommand/CommandSyntax.kt
package com.islandstudio.neon.stable.primary.nCommand

import org.bukkit.ChatColor

enum class CommandSyntax(val syntaxMessage: String) {
    INVALID_ARGUMENT("${NCommand.getPluginName()} ${ChatColor.RED}Invalid or missing argument!"),
    INVALID_PERMISSION("${NCommand.getPluginName()} ${ChatColor.RED}You don't have permissicon to use this command!"),
    INVALID_RANK("${NCommand.getPluginName()} ${ChatColor.RED}You need higher rank to use this command!");

    companion object {
        /**
         * Create custom syntax message
         *
         * @param syntaxMessage The custom syntax message including the plugin name
         */
        fun createSyntaxMessage(syntaxMessage: String): String {
            return "${NCommand.getPluginName()} $syntaxMessage"
        }
    }
}

